Improve memory utilization tests

yinan1048576 opened the pull request:
Added tests for large size classes and expanded the tests to
cover wider range of allocation sizes.

interwq commented on test/unit/extent_util.c:
```diff
@@ -25,18 +25,26 @@
-	assert_ptr_not_null(out, "test output allocation failed");
-	assert_ptr_not_null(out_ref, "test reference output allocation failed");
+	size_t sz;
+	for (sz = 7; sz <= TEST_MAX_SIZE && sz <= SC_LARGE_MAXCLASS;
```
nit: a bit comment about why / how the range is picked.

interwq commented on test/unit/extent_util.c:
```diff
@@ -46,55 +54,94 @@ TEST_BEGIN(test_query) {
+				    "Extent utilized count exceeded "
+				    "bin utilized count");
+			} else {
+				assert_zu_eq(BIN_NFREE_READ(out), 0,
```
I'd suggest not test this case (!config_stats) at all. We may want to avoid initializing the memory; or we can have the API to return EFAULT when stats is off.

interwq commented on test/unit/extent_util.c:
```diff
@@ -104,42 +151,50 @@ TEST_BEGIN(test_query) {
-	    "*oldlenp and newlen do not match");
-	in_sz += sizeof(const void *);
+	size_t sz;
+	for (sz = 17; sz <= TEST_MAX_SIZE && sz <= SC_LARGE_MAXCLASS;
```
Similarly, comments here.

yinan1048576 commented:
Revised according to feedback -
- Added more comments;
- Removed stats-specific checks when stats are disabled;
- Also modified the `mallctl` comments accordingly.

interwq commented on test/unit/extent_util.c:
```diff
@@ -25,18 +25,29 @@
-	assert_ptr_not_null(out, "test output allocation failed");
-	assert_ptr_not_null(out_ref, "test reference output allocation failed");
+	size_t sz;
+	/* Select some sizes that can span both small and large sizes, and are
```
nit: multi-line comments; first line with `/*` alone.

interwq commented on test/unit/extent_util.c:
```diff
@@ -104,42 +151,53 @@ TEST_BEGIN(test_query) {
-	    "*oldlenp and newlen do not match");
-	in_sz += sizeof(const void *);
+	size_t sz;
+	/* Select some sizes that can span both small and large sizes, and are
```
here as well.

yinan1048576 commented:
Fixed comments
